server/src/routes/trpc.ts

```diff
@@ -0,0 +1,57 @@
+import { initTRPC } from "@trpc/server";
+import { createExpressMiddleware } from "@trpc/server/adapters/express";
+import { z } from "zod";
+import { getDb } from "~/db/lib/get-db";
+
+const createContext = () => {
+  return {
+    ...getDb(process.env.DATABASE_URL!),
+  };
+};
+
+const t = initTRPC.context<typeof createContext>().create();
+
+const router = t.router;
+
+const trpcRouter = router({
+  getStudentById: t.procedure
+    .input(z.object({ student_id: z.string().uuid() }))
+    .query(async (req) => {
+      const { student_id } = req.input;
+
+      const result = await req.ctx.db
+        .selectFrom("student")
+        .where("student_id", "=", student_id)
+        .selectAll()
+        .executeTakeFirst();
+
+      return result;
+    }),
+
+  getAllStudents: t.procedure.query(async (req) => {
+    const result = await req.ctx.db.selectFrom("student").selectAll().execute();
+
+    return result;
+  }),
+
+  createStudent: t.procedure
```

Comment:
Related to the type comment above - what is the best practice for define the interface of request and return objects in trpc? In grpc, there is typically a Request and Response message that defines the input/output of a method. This method has only two input params so this is ok, but for some of the more complicated flows, we can easily have a dozen or more input params that we need the client to send.

Reply:
the request interface is defined by the `.input(...)` builder function with the Zod schema

response interfaces aren't currently explicitly defined but can be done similarly with `.output(...)`